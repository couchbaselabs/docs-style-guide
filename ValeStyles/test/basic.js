// this test file is designed to be run with Mocha
const assert = require('assert')
const fs = require('fs')
const ok = specify
const { spawnSync } = require('node:child_process')

console.log(process.cwd())

describe('run vale against test files', function () {

  let vale
  try {
    vale = spawnSync(
      'vale',
      [
        '--output', 'JSON',
        'test/fixtures/'
      ]
    )
  }
  catch (err) {
    console.log("Failed to run vale", err)
  }

  const output = JSON.parse(vale.stdout)

  for (const [file, actual] of Object.entries(output)) {
    ok(`testing ${file}`, function () {

      for (const item of actual) {
        delete item.Line
        delete item.Span
      }

      fs.writeFileSync(`${file}.actual`, JSON.stringify(actual, null, 2), 'utf8')
      
      try {
        const expected = JSON.parse(fs.readFileSync(`${file}.expected`, 'utf8'))
        assert.deepEqual(actual, expected)
      }
      catch (err) {
        if (err.code === 'ENOENT') {
          assert.fail(`No expected file found for ${file}, skipping comparison.`)
        } else {
          throw err
        }
      }
    })
  }
})


