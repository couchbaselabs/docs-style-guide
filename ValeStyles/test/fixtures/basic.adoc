= This is a basic test

This is a aardvark. He likes couchbase.
You are going to love this!
