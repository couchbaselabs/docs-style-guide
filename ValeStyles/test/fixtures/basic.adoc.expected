[
  {
    "Action": {
      "Name": "",
      "Params": null
    },
    "Check": "Couchbase.Headings",
    "Description": "",
    "Link": "",
    "Message": "'This is a basic test' should use title-style capitalization.",
    "Severity": "warning",
    "Match": "This is a basic test"
  },
  {
    "Action": {
      "Name": "",
      "Params": null
    },
    "Check": "Couchbase.An",
    "Description": "",
    "Link": "https://docs.couchbase.com/styleguide/articles.html",
    "Message": "Use \"an,\" not \"a.\"",
    "Severity": "error",
    "Match": "a aardvark"
  },
  {
    "Action": {
      "Name": "",
      "Params": null
    },
    "Check": "Couchbase.VentilatedProse",
    "Description": "",
    "Link": "",
    "Message": "Use ventilated prose. Start every sentence on a new line.",
    "Severity": "warning",
    "Match": "k. H"
  },
  {
    "Action": {
      "Name": "",
      "Params": null
    },
    "Check": "Vale.Terms",
    "Description": "",
    "Link": "",
    "Message": "Use 'Couchbase' instead of 'couchbase'.",
    "Severity": "error",
    "Match": "couchbase"
  },
  {
    "Action": {
      "Name": "replace",
      "Params": [
        "you're"
      ]
    },
    "Check": "Couchbase.Contractions",
    "Description": "",
    "Link": "https://docs.couchbase.com/styleguide/contractions.html",
    "Message": "Use 'you're' instead of 'You are'.",
    "Severity": "error",
    "Match": "You are"
  },
  {
    "Action": {
      "Name": "",
      "Params": null
    },
    "Check": "Google.Exclamation",
    "Description": "",
    "Link": "https://developers.google.com/style/exclamation-points",
    "Message": "Don't use exclamation points in text.",
    "Severity": "error",
    "Match": "s!"
  }
]